src/Pomelo.EntityFrameworkCore.MySql/Update/Internal/MySqlBatchExecutor.cs

```diff
@@ -4,120 +4,84 @@
 using Microsoft.EntityFrameworkCore.Storage;
-using Pomelo.Data.MySql;
 
 namespace Microsoft.EntityFrameworkCore.Update.Internal
 {
+  /// <summary>
+  ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+  ///     directly from your code. This API may change or be removed in future releases.
+  /// </summary>
+  public class MySqlBatchExecutor : IBatchExecutor
+  {
     /// <summary>
-    ///     This API supports the Entity Framework Core infrastructure and is not intended to be used 
+    ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
     ///     directly from your code. This API may change or be removed in future releases.
     /// </summary>
-    public class MySqlBatchExecutor : IBatchExecutor
+    public virtual int Execute(
+      IEnumerable<ModificationCommandBatch> commandBatches,
+      IRelationalConnection connection)
     {
-        /// <summary>
-        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used 
-        ///     directly from your code. This API may change or be removed in future releases.
-        /// </summary>
-        public virtual int Execute(
-            IEnumerable<ModificationCommandBatch> commandBatches,
-            IRelationalConnection connection)
+      var rowsAffected = 0;
+      connection.Open();
+      IDbContextTransaction startedTransaction = null;
+      try
+      {
+        if (connection.CurrentTransaction == null)
         {
-            var rowsAffected = 0;
-            connection.Open();
-            IDbContextTransaction startedTransaction = null;
-            try
-            {
-                if (connection.CurrentTransaction == null)
-                {
-                    startedTransaction = connection.BeginTransaction();
-                }
+          startedTransaction = connection.BeginTransaction();
+        }
 
-                foreach (var commandbatch in commandBatches)
-                {
-                    commandbatch.Execute(connection);
-                    // Fixed Issue #1: DataReader conflicted when added multiple entities
-                    try
-                    {
-                        lock (this)
-                        {
-                            var reader = (connection.DbConnection as MySqlConnection).Reader.LastOrDefault();
-                            if (reader != null)
-                            {
-                                reader.Dispose();
-                                (connection.DbConnection as MySqlConnection).Reader.Remove(reader);
-                            }
-                        }
-                    }
-                    catch
-                    {
-                    }
-                    rowsAffected += commandbatch.ModificationCommands.Count;
-                }
+        foreach (var commandbatch in commandBatches)
+        {
+            commandbatch.Execute(connection);
```

Comment:
need to reimplement locking here, so that one connection may only execute one statement at a time